### ISSUE TITLE
 bug: improve ssd1305 library drawing performance

### ISSUE THREAD

**Issue by jmamma** [jmamma/MIDICtrl20_MegaCommand#77]
7c58e3f40448655e13a679439d52446fde03b4fc

@yatli Could be a memory leak somewhere in this one.
WavDesigner was working fine before this commit.

In OscPage

including:

265   oled_display.fillRect(n + x, 0, scanline_width, 32, BLACK);

Causes trig interface to fail. and there is a weird display flicker.

--
```
248 void OscPage::draw_wav(uint8_t wav_type) {
249 #ifdef OLED_DISPLAY
250   uint8_t x = 64;
251   uint8_t y = 0;
252   uint8_t h = 30;
253   uint8_t w = 128 - x;
254   float osc_width = get_width();
255   TriOsc tri_osc(w, osc_width);
256   PulseOsc pul_osc(w, osc_width);
257   SawOsc saw_osc(w, osc_width);
258   SineOsc sine_osc(w);
259   UsrOsc usr_osc(w);
260   float sample;
261   float max_sine_gain = (float)1 / (float)16;
262   uint8_t n = sample_number;
263   // for (uint8_t n = 0; n < 128 - x; n++) {
264 
265   oled_display.fillRect(n + x, 0, scanline_width, 32, BLACK);
266   for (uint8_t n = sample_number; n < sample_number + scanline_width; n++) {
267     //  if ((scanline_width < w) && (wav_type > 0)) {
268     //  oled_display.drawLine(n + x, 0, n + x, 32, BLACK);
```

**Comment by jmamma** [comment]
Ah I think it's just overflow.

If rect.x + rect.w > 128 there is a problem.

**Comment by jmamma** [comment]
fixed: c489d15c9de092734dcd59e760aa423a1910cf31
